fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ElectionHandlerTest.java

```diff
@@ -74,228 +99,255 @@ public void setup() throws GeneralSecurityException, IOException {
+  private void handleElectionKey(Election election, String key)
+      throws UnknownElectionException, UnknownRollCallException, UnknownLaoException,
+          DataHandlingException, NoRollCallException {
+    // Create the election key message
+    ElectionKey electionKey = new ElectionKey(election.getId(), key);
 
-    assertEquals(EventState.CREATED, electionOpt.get().getState());
-    assertEquals(electionSetupOpenBallot.getId(), electionOpt.get().getId());
+    MessageGeneral message = new MessageGeneral(SENDER_KEY, electionKey, gson);
+    messageHandler.handleMessage(messageSender, election.getChannel(), message);
+  }
 
-    // Check that the election version has been successfully set
-    assertEquals(ElectionVersion.OPEN_BALLOT, electionOpt.get().getElectionVersion());
+  private void handleElectionOpen(Election election)
+      throws UnknownElectionException, UnknownRollCallException, UnknownLaoException,
+          DataHandlingException, NoRollCallException {
+    OpenElection openElection = new OpenElection(LAO.getId(), election.getId(), OPENED_AT);
 
-    // Check the WitnessMessage has been created
-    Optional<WitnessMessage> witnessMessage =
-        laoRepo.getLaoByChannel(LAO_CHANNEL).getWitnessMessage(message.getMessageId());
-    assertTrue(witnessMessage.isPresent());
+    MessageGeneral message = new MessageGeneral(SENDER_KEY, openElection, gson);
+    messageHandler.handleMessage(messageSender, election.getChannel(), message);
+  }
 
-    // Check the Witness message contains the expected title and description
-    WitnessMessage expectedMessage =
-        electionSetupWitnessMessage(message.getMessageId(), electionOpt.get());
-    assertEquals(expectedMessage.getTitle(), witnessMessage.get().getTitle());
-    assertEquals(expectedMessage.getDescription(), witnessMessage.get().getDescription());
+  private MessageID handleCastVote(ElectionVote vote, KeyPair senderKey)
+      throws UnknownElectionException, UnknownRollCallException, UnknownLaoException,
+          DataHandlingException, NoRollCallException {
+    CastVote<ElectionVote> electionVote =
+        new CastVote<>(
+            Collections.singletonList(vote), OPEN_BALLOT_ELECTION.getId(), CREATE_LAO.getId());
+
+    MessageGeneral message = new MessageGeneral(senderKey, electionVote, gson);
+    messageHandler.handleMessage(messageSender, OPEN_BALLOT_ELECTION.getChannel(), message);
+    return message.getMessageId();
   }
 
-  @Test
-  public void testHandleElectionResult()
-      throws DataHandlingException, UnknownLaoException, UnknownRollCallException,
-          NoRollCallException {
-    // Create the result Election message
-    QuestionResult questionResult =
-        new QuestionResult(electionQuestion.getBallotOptions().get(0), 2);
+  // Once ElectionVote and EncryptedVote have the same parent, merge the two functions and use the
+  // parent
+  private MessageID handleCastVote(ElectionEncryptedVote vote, KeyPair senderKey)
```

Comment:
It is not immediately clear imo that this comment is meant for future work and not describing what the function do